pyincore/analyses/indp/dislocationutils.py

```diff
@@ -0,0 +1,158 @@
+                    # Find building in the service area of the pipe
+                    serv_area = mapping_data[mapping_data['edge_guid'] == guid]
+                    start_node = u[0]
+                    if start_node not in node_pop.keys():
+                        node_pop[start_node] = {'total_pop_node': 0,
+                                                'num_dilocated': {t: 0 for t in range(T + 1)}}
+                    end_node = v[0]
+                    if end_node not in node_pop.keys():
+                        node_pop[end_node] = {'total_pop_node': 0,
+                                              'num_dilocated': {t: 0 for t in range(T + 1)}}
+                    # compute dynamic_params
+                    for _, bldg in serv_area.iterrows():
+                        pop_bldg_dict = pop_dislocation[pop_dislocation['guid'] == bldg['bldg_guid']]
+                        for _, hh in pop_bldg_dict.iterrows():
+                            # half of the arc's demand is assigned to each node
+                            # also, each arc is counted twice both as (u,v) and (v,u)
+                            node_pop[start_node]['total_pop_node'] += hh['numprec'] / 4 \
+                                if ~np.isnan(hh['numprec']) else 0
+                            node_pop[end_node]['total_pop_node'] += hh['numprec'] / 4 \
+                                if ~np.isnan(hh['numprec']) else 0
+                            if hh['dislocated']:
+                                # ..todo Lumebrton dislocation time model. Replace with that of Seaside when available
+                                return_time = DislocationUtil.lumberton_disloc_time_mode(hh)
+                                for t in range(return_time):
+                                    if t <= T and return_type == 'step_function':
+                                        node_pop[start_node]['num_dilocated'][t] += hh['numprec'] / 4 \
+                                            if ~np.isnan(hh['numprec']) else 0
+                                        node_pop[end_node]['num_dilocated'][t] += hh['numprec'] / 4 \
+                                            if ~np.isnan(hh['numprec']) else 0
+                                    elif t <= T and return_type == 'linear':
+                                        pass  # ..todo Add linear return here
+                for n, val in node_pop.items():
+                    for t in range(T + 1):
+                        values = [t, n, val['total_pop_node'] - val['num_dilocated'][t],
+                                  val['total_pop_node']]
+                        dynamic_params[nn] = dynamic_params[nn].append(dict(zip(dp_dict_col, values)),
+                                                                       ignore_index=True)
+        with open(output_file, 'wb') as f:
+            pickle.dump(dynamic_params, f)
+        return dynamic_params
+
+    @staticmethod
+    def lumberton_disloc_time_mode(household_data):
```

Comment:
is this specific to lumberton?

Reply:
oh noes i didn't notice this module... 
this looks like something we should separate out. Maybe in the next release?

Reply:
for now i can rename this just `disloc_time_mode`?

Reply:
Sounds good. Refactoring next release works so we can get this in.